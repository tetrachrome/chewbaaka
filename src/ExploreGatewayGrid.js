/**
 * ExploreGatewayGrid.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 02, 2020
 * Updated  : Jul 02, 2020
 */

import React from 'react';

import "semantic-ui-css/semantic.min.css";

import { Grid } from 'semantic-ui-react'

import ExploreGatewayGridItem from './ExploreGatewayGridItem.js'

function ExploreGatewayGrid(props) {
  return (
    <Grid container stackable textAlign="center" columns={4}>
      <ExploreGatewayGridItem imageUrl="https://calacademy.org/sites/all/themes/calacademy_zen/images/homepage/slider-aquarium.jpg" title="History" />
      <ExploreGatewayGridItem imageUrl="https://calacademy.org/sites/all/themes/calacademy_zen/images/homepage/slider-aquarium.jpg" title="Biology" />
      <ExploreGatewayGridItem imageUrl="https://calacademy.org/sites/all/themes/calacademy_zen/images/homepage/slider-aquarium.jpg" title="Ecology" />
      <ExploreGatewayGridItem imageUrl="https://calacademy.org/sites/all/themes/calacademy_zen/images/homepage/slider-aquarium.jpg" title="Future" />
    </Grid>
  )
}

export default ExploreGatewayGrid;
