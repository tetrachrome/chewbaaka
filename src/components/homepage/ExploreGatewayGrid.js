/**
 * ExploreGatewayGrid.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 02, 2020
 * Updated  : Jul 13, 2020
 */

import React from 'react';

import "semantic-ui-css/semantic.min.css";

import { Grid } from 'semantic-ui-react'

import ExploreGatewayGridItem from './ExploreGatewayGridItem'

import explore_gateway_grid_item_img_01 from './assets/explore_gateway_grid_item_img_01.jpg'
import explore_gateway_grid_item_img_02 from './assets/explore_gateway_grid_item_img_02.jpg'
import explore_gateway_grid_item_img_03 from './assets/explore_gateway_grid_item_img_03.jpg'
import explore_gateway_grid_item_img_04 from './assets/explore_gateway_grid_item_img_04.jpg'

export default function ExploreGatewayGrid(props) {
  return (
    <Grid container stackable textAlign="center" columns={4} data-testid="ExploreGatewayGridTestId">
      <ExploreGatewayGridItem image={explore_gateway_grid_item_img_01} title="History" dstUrl="history" />
      <ExploreGatewayGridItem image={explore_gateway_grid_item_img_02} title="Biology" dstUrl="biology" />
      <ExploreGatewayGridItem image={explore_gateway_grid_item_img_03} title="Ecology" dstUrl="ecology" />
      <ExploreGatewayGridItem image={explore_gateway_grid_item_img_04} title="Future"  dstUrl="future" />
    </Grid>
  )
}
