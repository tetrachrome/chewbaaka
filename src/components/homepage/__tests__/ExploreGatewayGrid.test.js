/**
 * ExploreGatewayGrid.test.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 04, 2020
 * Updated  : Jul 04, 2020
 */

import React from 'react';

import { render, screen } from '@testing-library/react';

import ExploreGatewayGrid from '../ExploreGatewayGrid';

test('renders explore gateway grid', () => {
  render(<ExploreGatewayGrid />);

  const expectedNumGridComponentChildren = 4;

  // Tests grid component has the expected number of children components present.
  const gridComponent = screen.getByTestId("ExploreGatewayGridTestId");
  expect(gridComponent.children.length).toBe(expectedNumGridComponentChildren);

  // Tests grid item components are present.
  const expectedGridItemTitles = [
    /History/i,
    /Biology/i,
    /Ecology/i,
    /Future/i,
  ];

  for ( const [_, expectedGridItemTitle] in expectedGridItemTitles.entries() ) {
    const gridItemComponentForTitle = screen.getByText(expectedGridItemTitle);
    expect(gridItemComponentForTitle).toBeInTheDocument();
  }
});
