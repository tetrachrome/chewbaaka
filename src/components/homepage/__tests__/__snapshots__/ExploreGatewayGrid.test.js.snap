// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExploreGatewayGrid component snapshot 1`] = `
<nav>
  <div
    className="ui container stackable center aligned four column grid"
    data-testid="ExploreGatewayGridTestId"
  >
    <div
      className="column ExploreGatewayGridItemColumn"
    >
      <div
        className="ExploreGatewayGridItemInnerContainer"
      >
        <a
          className="ExploreGatewayGridItemImageAnchor"
          data-testid="ExploreGatewayGridItemImageAnchorTestId"
          href="/history"
          onClick={[Function]}
        >
          <img
            alt="History"
            className="ui medium image ExploreGatewayGridItemImage"
            data-testid="ExploreGatewayGridItemImageTestId"
            src="explore_gateway_grid_item_img_01-min.jpg"
          />
        </a>
        <h2
          className="ui left aligned header ExploreGatewayGridItemHeader"
        >
          History
        </h2>
      </div>
    </div>
    <div
      className="column ExploreGatewayGridItemColumn"
    >
      <div
        className="ExploreGatewayGridItemInnerContainer"
      >
        <a
          className="ExploreGatewayGridItemImageAnchor"
          data-testid="ExploreGatewayGridItemImageAnchorTestId"
          href="/biology"
          onClick={[Function]}
        >
          <img
            alt="Biology"
            className="ui medium image ExploreGatewayGridItemImage"
            data-testid="ExploreGatewayGridItemImageTestId"
            src="explore_gateway_grid_item_img_02-min.jpg"
          />
        </a>
        <h2
          className="ui left aligned header ExploreGatewayGridItemHeader"
        >
          Biology
        </h2>
      </div>
    </div>
    <div
      className="column ExploreGatewayGridItemColumn"
    >
      <div
        className="ExploreGatewayGridItemInnerContainer"
      >
        <a
          className="ExploreGatewayGridItemImageAnchor"
          data-testid="ExploreGatewayGridItemImageAnchorTestId"
          href="/ecology"
          onClick={[Function]}
        >
          <img
            alt="Ecology"
            className="ui medium image ExploreGatewayGridItemImage"
            data-testid="ExploreGatewayGridItemImageTestId"
            src="explore_gateway_grid_item_img_03-min.jpg"
          />
        </a>
        <h2
          className="ui left aligned header ExploreGatewayGridItemHeader"
        >
          Ecology
        </h2>
      </div>
    </div>
    <div
      className="column ExploreGatewayGridItemColumn"
    >
      <div
        className="ExploreGatewayGridItemInnerContainer"
      >
        <a
          className="ExploreGatewayGridItemImageAnchor"
          data-testid="ExploreGatewayGridItemImageAnchorTestId"
          href="/future"
          onClick={[Function]}
        >
          <img
            alt="Future"
            className="ui medium image ExploreGatewayGridItemImage"
            data-testid="ExploreGatewayGridItemImageTestId"
            src="explore_gateway_grid_item_img_04-min.jpg"
          />
        </a>
        <h2
          className="ui left aligned header ExploreGatewayGridItemHeader"
        >
          Future
        </h2>
      </div>
    </div>
  </div>
</nav>
`;
